backend/infrahub/graphql/mutations/main.py

```diff
@@ -160,6 +163,88 @@ async def _call_mutate_create_object(cls, data: InputObjectType, db: InfrahubDat
+    ) -> None:
+        component_registry = get_component_registry()
+        node_constraint_runner = await component_registry.get_component(NodeConstraintRunner, db=db, branch=branch)
+
+        for relationship in obj.get_schema().relationships:
+            if relationship.name in data or relationship.kind != RelationshipKind.COMPONENT:
+                continue
+
+            template_relationship_manager: RelationshipManager = getattr(template, relationship.name)
+            if relationship.cardinality == RelationshipCardinality.MANY:
+                template_relationship_peers = await template_relationship_manager.get_peers(db=db)
+            else:
+                template_relationship_peers = {}
+                if template_relationship_peer := await template_relationship_manager.get_peer(db=db):
+                    template_relationship_peers[template_relationship_peer.id] = template_relationship_peer
+
+            if not template_relationship_peers:
+                continue
+
+            obj_peer_schema = (
+                template_relationship_manager.schema.get_peer_schema(db=db, branch=branch)
+                .get_relationship("related_nodes")
+                .get_peer_schema(db=db, branch=branch)
+            )
+
+            for template_relationship_peer in template_relationship_peers.values():
+                obj_peer_data: dict[str, Any] = {}
+
+                for relationship_peer_attr in template_relationship_peer._attributes:
+                    if relationship_peer_attr not in obj_peer_schema.attribute_names:
+                        continue
+                    obj_peer_data[relationship_peer_attr] = {
+                        "value": getattr(template_relationship_peer, relationship_peer_attr).value
+                    }
+
+                for relationship_peer_rel in template_relationship_peer._relationships:
+                    relationship_peer_rel_value: RelationshipManager = getattr(
+                        template_relationship_peer, relationship_peer_rel
+                    )
+                    if (
+                        relationship_peer_rel_value.schema.kind
+                        not in [RelationshipKind.COMPONENT, RelationshipKind.PARENT]
+                        or relationship_peer_rel_value.schema.name not in obj_peer_schema.relationship_names
+                    ):
+                        continue
+
+                    v = await relationship_peer_rel_value.get_peers(db=db)
+                    if list(v) == [template.id]:
+                        obj_peer_data[relationship_peer_rel] = {"id": obj.id}
+
+                obj_peer = await Node.init(schema=obj_peer_schema, db=db)
+
+                await obj_peer.new(db=db, **obj_peer_data)
+                await node_constraint_runner.check(node=obj_peer, field_filters=list(obj_peer_data))
+                await obj_peer.save(db=db)
```

Comment:
When saving the peer here we would have a changelog for that peer under `obj_peer.node_changelog` I think we'd want the .handle_relationships() method to send back a `list[NodeChangelog]` to the caller in order to get a full picture of what has changed.